maven-settings/src/main/java/org/apache/maven/settings/io/xpp3/SettingsXpp3Reader.java

```diff
@@ -49,98 +52,101 @@ public SettingsXpp3Reader(ContentTransformer contentTransformer) {
      * @return boolean
      */
     public boolean getAddDefaultEntities() {
-        return addDefaultEntities;
-    } // -- boolean getAddDefaultEntities()
+        return delegate.getAddDefaultEntities();
+    }
+
+    /**
+     * Sets the state of the "add default entities" flag.
+     *
+     * @param addDefaultEntities a addDefaultEntities object.
+     */
+    public void setAddDefaultEntities(boolean addDefaultEntities) {
+        delegate.setAddDefaultEntities(addDefaultEntities);
+    }
 
     /**
      * @param reader a reader object.
      * @param strict a strict object.
      * @throws IOException IOException if any.
-     * @throws XMLStreamException XMLStreamException if
+     * @throws XmlPullParserException XmlPullParserException if
      * any.
      * @return Settings
      */
-    public Settings read(Reader reader, boolean strict) throws IOException, XMLStreamException {
-        XMLInputFactory factory = new com.ctc.wstx.stax.WstxInputFactory();
-        factory.setProperty(XMLInputFactory.IS_REPLACING_ENTITY_REFERENCES, false);
-        XMLStreamReader parser = null;
+    public Settings read(Reader reader, boolean strict) throws IOException, XmlPullParserException {
         try {
-            parser = factory.createXMLStreamReader(reader);
+            return new Settings(delegate.read(reader, strict, null));
         } catch (XMLStreamException e) {
-            throw new RuntimeException(e);
+            throw new XmlPullParserException(e.getMessage(), null, e);
         }
-        return read(parser, strict);
-    } // -- Model read( Reader, boolean )
+    }
 
     /**
      * @param reader a reader object.
      * @throws IOException IOException if any.
-     * @throws XMLStreamException XMLStreamException if
+     * @throws XmlPullParserException XmlPullParserException if
      * any.
      * @return Model
      */
-    public Settings read(Reader reader) throws IOException, XMLStreamException {
-        return read(reader, true);
-    } // -- Model read( Reader )
+    public Settings read(Reader reader) throws IOException, XmlPullParserException {
+        try {
+            return new Settings(delegate.read(reader));
+        } catch (XMLStreamException e) {
+            throw new XmlPullParserException(e.getMessage(), null, e);
```

Comment:
Can't we just pass the exception?

Reply:
> Can't we just pass the exception?

Pass how / where ?  We can't pass through, that would change the `throws` clause and break the API.  Which would be problematic because that's the whole purpose of that class.

Reply:
That is not what I meant. Single arg ctor.

Reply:
Yeah, another one is missing for this case

Reply:
I think the xpp3 api may even predate chained exceptions that were introduced in JDK 1.4 ;-)